<launch>

	<node pkg="tf" type="static_transform_publisher" name="hermes_camera" args="0 0 0.1016 0.5 0.5 0.5 0.5 hermes hermes/camera 100" />

	<node name="hermes_node" pkg="hermes" type="hermes_node" output="screen" ns="hermes"/>

	<!--Navigation stack setup-->
	<node name="ar_localization" pkg="navigation" type="ar_tag_localization" output="screen" ns="hermes">
		<param name="target_frame" value="hermes" />
		<param name="filtered_odom_frame" value="hermes/odom_combined" />
		<param name="wheel_odom_frame" value="hermes/wheel_odom" />
	</node>
	
	<node name="odom_filter" pkg="robot_pose_ekf" type="robot_pose_ekf" respawn="true" ns="hermes">
		<param name="output_frame" value="odom_combined"/>
		<param name="base_footprint_frame" value="map" />
    	<param name="freq" value="10.0"/>
    	<param name="sensor_timeout" value=".5"/>
    	<param name="odom_used" value="true"/>
    	<param name="imu_used" value="false"/>
    	<param name="vo_used" value="true"/>
    	<param name="debug" value="true"/>
    	<param name="self_diagnose" value="true"/>
    	<remap from="odom" to="wheel_odom" />
    	<remap from="vo" to="ar_tag_odom" />
  </node>

	<node name="hermes_planner" pkg="navigation" type="astar" output="screen" ns="hermes">
		<param name="pose_topic" value="pose" />
		<param name="map_topic" value="/hsmrs/map" />
	</node>

	<node name="hermes_executor" pkg="navigation" type="path_executor" output="screen" ns="hermes">
		<param name="pose_topic" value="pose" />
		<param name="vel_topic" value="cmd_vel_mux/input/teleop" />
		<param name="max_linear_velocity" value="0.2" />
		<param name="max_angular_velocity" value="0.5" />		
	</node>

	<!--AR Marker setup-->
	<arg name="nav_marker_size" default="20.3" />
	<arg name="obj_marker_size" default="10" />
	<arg name="max_new_marker_error" default="0.08" />
	<arg name="max_track_error" default="0.1" />
	<arg name="cam_image_topic" default="/hermes/camera" />
	<arg name="cam_info_topic" default="/hermes/camera_info" />
	<arg name="output_frame" default="/base_link" />

	<node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen" args="$(arg nav_marker_size) $(arg max_new_marker_error) $(arg max_track_error) $(arg cam_image_topic) $(arg cam_info_topic) $(arg output_frame)" ns="hermes"/>

	<!--node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen" args="$(arg obj_marker_size) $(arg max_new_marker_error) $(arg max_track_error) $(arg cam_image_topic) $(arg cam_info_topic) $(arg output_frame)" ns="hermes"/-->

</launch>
